src/org/zaproxy/zap/extension/ascan/ActiveScan.java

```diff
@@ -83,8 +84,8 @@
 	private static final Logger log = Logger.getLogger(ActiveScan.class);
 
 	public ActiveScan(String displayName, ScannerParam scannerParam, 
-			ConnectionParam param, ScanPolicy scanPolicy) {
-		super(scannerParam, param, scanPolicy);
+			ConnectionParam param, ScanPolicy scanPolicy, RuleConfigParam ruleConfigParam) {
+		super(scannerParam, param, scanPolicy, ruleConfigParam);

```

Comment:
Other classes would have to be changed too (e.g. HostProcess, Scanner...), IMHO these classes shouldn't be used directly by 3rd-party code (including zap-extensions add-ons).
I guess we need to define what belongs to the public API and what not.